src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -89,6 +89,22 @@ var DOMRenderer = ReactFiberReconciler({
     return true;
   },
 
+  shouldSetTextContent(props : Props) : boolean {
+    return (
+      typeof props.children === 'string' ||
+      typeof props.children === 'number' ||
+      (
+        typeof props.dangerouslySetInnerHTML === 'object' &&
+        props.dangerouslySetInnerHTML !== null &&
+        typeof props.dangerouslySetInnerHTML.__html === 'string'
+      )
+    );
+  },
+
+  resetTextContent(domElement : Instance) : void {
+    domElement.textContent = '';
```

Comment:
If we're concerned with performance then IIRC setting textContent or innerHTML is unintuitively significantly slower than simply clearing the children one by one.